src/app/static/src/app/components/Warning.vue

```diff
@@ -0,0 +1,173 @@
+                fullBoxHeight: 0,
+                lineHeight: 0
+            };
+        },
+        computed: {
+            renderedBoxHeight(){
+                if (!this.warningsLengthy){
+                    return this.fullBoxHeight
+                } else {
+                    return this.showFullBox ? this.fullBoxHeight : this.maxBoxHeight
+                }
+            },
+            maxBoxHeight(){
+                return this.maxLines * this.lineHeight
+            },
+            warningsLengthy(){
+                return this.fullBoxHeight > this.maxBoxHeight
+            },
+            currentLanguage: mapStateProp<RootState, Language>(
+                null,
+                (state: RootState) => state.language
+            ),
+            buttonText(){
+                return i18next.t(this.showFullBox ? "showLess" : "showMore", { lng: this.currentLanguage });
+            },
+            // The user has the option to configure the number of lines that will be displayed. 
+            // If the height of the warnings rendered are more than this number, each warning 
+            // will be truncated to one line and the show more button will appear. If there is 
+            // only one warning message, it will be truncated after the max number of lines
+            lineStyling(){
+                if (this.warningsLengthy && !this.showFullBox){
+                    if (this.warnings.length === 1){
+                        return {
+                            height: `${this.maxBoxHeight}px`,
+                            overflow: "hidden",
+                            "display": "-webkit-box",
+                            "-webkit-line-clamp": this.maxLines,
+                            "-webkit-box-orient": "vertical"
+                        }
+                    } else {
+                        return {
+                            height: `${this.lineHeight}px`,
+                            whiteSpace: "nowrap",
+                            overflow: "hidden",
+                            textOverflow: "ellipsis"
+                        }
+
+                    }
+                } else {
+                    return {}
+                }
+            }
+        },
+        methods: {
+            toggleShowFullBox(){
+                this.showFullBox = !this.showFullBox;
+                if (this.showFullBox){
+                    this.updateDimensions();
+                }
+            },
+            updateDimensions(){
+                let counter = 0
+                const id = setInterval(() => {
+                    counter++
+                    if (this.$refs.warningBox){
+                        this.lineHeight = (this.$refs.line as HTMLElement).clientHeight;
+                        this.fullBoxHeight = (this.$refs.warningBox as HTMLElement).clientHeight;
+                        clearInterval(id)
+                    } else if (counter > 20){
+                        clearInterval(id)
+                    }
```

Comment:
Perhaps unnecessary, but I added a counter to prevent this setInterval persisting.

Reply:
Sorry, I'm confused about the need for this `setInterval` - what is it for? It looks like it's waiting for `$refs.warningBox` to exist, but shouldn't that always exist (as long as you have any warnings in the prop)?

Do you just need to make this async and add `await Vue.nextTick()` to let the DOM catch up following the toggle of `showFullBox`? I tried it that way locally and everything seemed as if it worked to me.